DiscountDiscover/View Controllers/DealsViewController.m

```diff
@@ -84,10 +84,9 @@ -(BOOL)tabBarController:(UITabBarController *)tabBarController shouldSelectViewC
     
     if ([viewController.restorationIdentifier isEqual: @"MapNavigationController"]) {
         UINavigationController *nc = (UINavigationController *)viewController;
-        MapViewController *vc = (MapViewController *)nc.topViewController;
+        MapViewController *vc = (MapViewController *)nc.viewControllers[0];
```

Comment:
This could cause an array out of bounds exception at runtime, and crash the app. I general you should always check before accessing an array index that it's within bounds. You can also use `NSArray`'s `firstObject` and `lastObject` methods, which will just return nil if the array is empty, but won't crash the app.